docs/requirements.txt

```diff
@@ -11,7 +11,7 @@ requests
 rich
 setuptools_scm
 simplejson
-sphinx<7.2
+sphinx
 sphinx-argparse
```

Comment:
Unrelated to this PR, but do we still need the `sphinx-argparse` plugin since we dropped the CLI docs?